discovery-provider/src/api/v1/helpers.py

```diff
@@ -499,6 +487,23 @@ def __schema__(self):
         return param
 
 
+# Helper to allow consumer to pass message and signature headers as request params
+def add_auth_headers_to_parser(parser):
```

Comment:
I don't know why I didn't think of doing this with helper methods, this is so smart 😂 

wayyy prefer this to my inheritance based approach for the other stuff